arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -670,3 +670,32 @@ def rule_gfm_1_3_8(
             msg=_("The submitter-specific taxonomy has an embedded linkbase."),
             modelObject=embeddedElements
         )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_5_6(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.5.6] The length of a label must be less than 511 characters unless it its role is documentation.
+    """
+    labelRelationshipSet = val.modelXbrl.relationshipSet(XbrlConst.conceptLabel)
+    if labelRelationshipSet is not None:
```

Comment:
Nit: Prefer "early exit" over nested if statements.
```suggestion
    if labelRelationshipSet is None:
    	return
```